core-bundle/src/Resources/contao/library/Contao/Folder.php

```diff
@@ -539,6 +545,52 @@ protected function getPathinfo()
 
 		return $return;
 	}
+
+	/**
+	 * Scan a directory and return its files and folders as array
+	 *
+	 * @param string  $strFolder
+	 * @param boolean $blnUncached
+	 *
+	 * @return array
+	 */
+	public static function scan($strFolder, $blnUncached=false): array
+	{
+		static::$arrScanCache;
+
+		// Add a trailing slash
+		if (substr($strFolder, -1, 1) != '/')
+		{
+			$strFolder .= '/';
+		}
+
+		// Load from cache
+		if (!$blnUncached && isset($arrScanCache[$strFolder]))
+		{
+			return $arrScanCache[$strFolder];
+		}
+
+		$arrReturn = array();
+
+		// Scan directory
+		foreach (scandir($strFolder, SCANDIR_SORT_ASCENDING) as $strFile)
+		{
+			if ($strFile == '.' || $strFile == '..')
+			{
+				continue;
+			}
+
+			$arrReturn[] = $strFile;
+		}
+
+		// Cache the result
+		if (!$blnUncached)
+		{
+			$arrScanCache[$strFolder] = $arrReturn;
+		}
+
+		return $arrReturn;
+	}
```

Comment:
Shouldn't we use the Symfony finder for this instead?

Reply:
Yes, we should. ☺️
But I think this is not required for this pull request. 🙃